mathesar_ui/src/stores/table-data/tabularData.ts

```diff
@@ -102,8 +105,12 @@ export class TabularData {
         ),
     );
 
+    const table = get(currentTable) as TableEntry;
```

Comment:
Relying on the `currentTable` store is problematic because it assumes that we'll never have an instance of `TabularData` for a table other than `currentTable`. One place where that assumption does not hold is the Record Page where we have one `TabularData` instance for the current table, plus any number of other instances for related tables too. The Record Selector also instantiates `TabularData` from the Schema Page when `$currentTable` is `undefined`.

To be fair, I would say that our current design of relying on this imported `currentTable` store (and others like it) is a deeper root cause of this problem because it leads people like you astray. We should aim for code that doesn't lead to these false assumptions. I'd rather we use imported stores only for things like "modal" and "toast" where we really _do_ expect a singleton (but that's a topic for another day).

For this PR, I think we should modify `TabularDataProps`, adding a new property like `table: TableEntry`. Then we would instantiate `TabularData` with a `TableEntry` everywhere. Searching our codebase for `new TabularData` seems to indicate that we're only instantiating `TabularData` four times, and in three of those cases we already have a `TableEntry` in scope. The exception is in `RecordSelectorWindow.svelte`, where it seems we could easily derive a `TableEntry` object from `tableId` (in scope) and the `tables` store (imported from `src/stores/tables.ts`).

@pavish How does the above paragraph sound to you. We would be making `TabularData` dependent on having a `TableEntry`. This would also allow us to persist the `TableEntry` within the `TabularData` instance, allowing us to see the table's name, for example. I have a vague memory that for some reason many months ago you didn't want to do this. If we don't want to require an entire `TableEntry` object, then we could pass in the minimum data needed here which would mean adding a property like `columnOrder: number[]` to `TabularDataProps`.

Reply:
I went ahead and implemented this solution. @pavish does it look ok to you? I can make changes to it if needed.

Reply:
@AlekLefebvre It's going to take me a while to get my head into this. We're currently focused on our initial release which should be by the end of this month.

While there's chances of merge conflicts, erring on the side of caution w.r.t regressions, I'd like to wait until the initial release is done and tagged, before merging this PR in. I'll take a look at this sometime next week.

cc: @seancolsen

Reply:
I looked through the code and the approach looks good to me.

I agree that a store restructuring is definitely required, and perhaps also change the way we access them since `tables.data.get(someId)` can be `undefined` and a request to fetch it may not be in progress. For now, I think no changes in approach is required in this PR.